Tests/ParameterEncoderTests.swift

```diff
@@ -485,6 +485,54 @@ final class URLEncodedFormEncoderTests: BaseTestCase {
         // Then
         XCTAssertFalse(result.isSuccess)
     }
+    
+    func testThatEncodableStructCanBeEncodedWithIndex() {
```

Comment:
Additionally, please add test cases for the other encodable types, like the struct, and the encodable subclass.